src/helpers/utils.ts

```diff
@@ -5,3 +5,31 @@ export const mod = (x: number, n: number) => ((x % n) + n) % n;
 
 export const getBorderColorByStatus = ({ hasError, isValid }: IInputStatus) =>
   hasError ? colors.alert : isValid ? colors.success : colors.greyLight;
+
+// @see https://github.com/PimpTrizkit/PJs/wiki/12.-Shade,-Blend-and-Convert-a-Web-Color-(pSBC.js)#stackoverflow-archive-begin
+export const shadeHexColor = (color: string, floatingPercent: number) => {
+  /**
+   * @param `floatingPercent` A number between -1 to 1. Positive is brighter and negative darker.
+   *                          To make a color 8% darker: -0.08;
+   *                          To make a color 50% lighter: 0.5;
+   */
+
+  const actualColor = parseInt(color.slice(1), 16);
+  const transparency = floatingPercent < 0 ? 0 : 255;
+  const actualPercent = floatingPercent < 0 ? floatingPercent * -1 : floatingPercent;
+  const r = actualColor >> 16;
+  const g = (actualColor >> 8) & 0x00ff;
+  const b = actualColor & 0x0000ff;
+
+  return (
+    '#' +
+    (
+      0x1000000 +
+      (Math.round((transparency - r) * actualPercent) + r) * 0x10000 +
+      Math.round((transparency - g) * actualPercent + g) * 0x100 +
+      (Math.round((transparency - b) * actualPercent) + b)
+    )
+      .toString(16)
+      .slice(1)
+  );
+};
```

Comment:
I have mixed feelings about this helper, happy to [just inline the hover hex colours](https://github.com/zopaUK/react-components/blob/537a6c52731642a3e1eac98ea52a33c937d467ca/src/components/atoms/Button/Button.tsx#L23-L33) if you think it's simpler 👍🏻 💆🏻‍♂️

Reply:
yep, looks like there is a lot of logic. Also, I don't understand what are you trying to achieve

Reply:
<img width="578" alt="Screenshot 2020-04-01 at 10 17 59" src="https://user-images.githubusercontent.com/5938217/78114815-22279f00-7402-11ea-84b0-fce3966a8ba2.png">

The _hover states_ are a lighter / darker version of a base colour. Hence the shade utility. I think though is too complicated and inlining the values will be better for now 👍🏻

Reply:
Agree